### PR TITLE
Updates README with Docker information

### PR DESCRIPTION
**Pull request by matsduf**
## Purpose

This PR

* Adds information on Zonemaster on Docker
* Adds table of contents
* Adds mentioning of Spanish
* Editorial updates

The updated document has a link to a document that exists in develop branch only. The link is on purpose to the master branch, where the document will be available after release.
```
https://github.com/zonemaster/zonemaster/blob/master/docs/internal-documentation/maintenance/ReleaseProcess-create-docker-image.md
```

The updated document has a link to a document that has been updated in the develop branch. The link is on purpose to the master branch, where the document will be available after release.
```
https://github.com/zonemaster/zonemaster-cli/blob/master/USING.md
```

## Changes

Updates README.md

## How to test this PR

Documentation only.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,28 +96,33 @@ and is available at large cloud providers.
...
 
 * FreeBSD uses MySQL, not MariaDB. 
 * FreeBSD bundles SQLite in Perl DBD::SQLite.
-
-Zonemaster Backend has been tested with the combination of OS and database
-engine version listed in the table above. Zonemaster uses functionality
-introduced in PostgreSQL version 10, and earlier versions are as such not supported.
+* Zonemaster Backend has been tested with the combination of OS and database
+  engine version listed in the table above.
+* Zonemaster uses functionality introduced in PostgreSQL version 10, and earlier
```
Comment (mattias-p) [inline README.md:108]: `uses` -> `depends on` maybe?

**Review by mattias-p — APPROVED** [review]
I really like the introduction of bullets :)

File: README.md (hunk 1 above)
Comment (matsduf) [reply README.md:108]: Fine for me. Updated.

**Review by mattias-p — APPROVED** [review]
LGTM

**Review by mattias-p — APPROVED** [review]

**Review by blacksponge — APPROVED** [review]
lgtm

**Pull request merged**